### PR TITLE
creating modules

### PR DESCRIPTION
**Pull request by DarrylD**

### CODE REVIEW

File: src/providers/peach/peach.js [hunk 1]
```diff
@@ -30,4 +32,7 @@ class PeachProvider {
 
 PeachProvider.$inject = [];
 
-export default PeachProvider;
+angular.module(moduleName, [])
```
Comment (jsmylnycky) [inline src/providers/peach/peach.js]: We would probably want to inject the peachAccount, peachUser, etc modules into here, right?

Comment (DarrylD) [inline src/providers/peach/peach.js]: Yep, need to be there to be truly encapsulated. Haven't gotten that far yet.

**Comment by jsmylnycky** [comment]
:+1:

**Comment by DarrylD** [comment]
@jsmylnycky @againsley @rynomars  Lets bring the conversation here :)

Any reason for that other that the fact that it's in his styleguide? I worked on projects with that structure, it made little sense for directives, a little more for actual modules (think a core app) but, keep in mind he isn't considering ES6 modules with Angular, just Angular modules. Abstracting the declaration seems to add more overhead opposed to "value add" and will start to resemble a file structure of a monolithic Java application.

Not to mention, as far as migration paths to 2.0 go, that `.module` file will probably be removed. As this structure resembles a web-component more.

Also, wouldn't I need to add `.directive` to the mix to keep consistent with the other files that have prefix 'types'?

**Comment by jsmylnycky** [comment]
@DarrylD Always up for a discussion :)

My initial feeling about it (and you did have me ok with the idea when we talked last week) was to lean towards a module file.  The reason being was that it makes it harder for us to declare multiple files and add them into the module if that scenario arises. Right now, outside of Directives (with their directive file and the controller file), that case isn't really popping up at all.  But that doesn't necessarily mean we wont ever have this pop up.  It feels funny to me to have to modify peach-event.js in order to add some other service to it, instead of just appending the service ie /code angular.module('$peachEvent').service('myAdditionalService');


If the solution is that every time theres a folder, that one singular file ends up being its own module, then I can agree with putting the module declaration in it at the bottom.  In the case of ngPeach where we have multiple modules being bundled in, and ngPeach is then a module that brings them all together.  If we want to follow this pattern, I'm ok with it too.  We definitely will need to make a decision one way or another on this and stick to it.

I'll let @rynomars  chime in on his thoughts.


As for the .directive addition to the pathname, I'm not opposed to it.  I could have sworn there was a rule on the styleguide that says that if the file type (in this case directive) was preceded by a directory named similarly, it wasn't necessary to include the .directive in the naming.  I'm struggling to find it, so maybe it was ruled out.  We definitely would want to decide one way or another which way we want to go with it tho.

**Comment by rynomars** [comment]
@DarrylD my only concern with the initial implementation is if we have another service/provider that we need to add to the module down the road.  If we do happen to add another service/provider, then we will need to move the line out of the file. 

You can only have the creation of the module once in your app. Any other call to 
`var myModule = angular.module('myModule', []);`
will overwrite any previous calls.

**Comment by DarrylD** [comment]
@jsmylnycky Ok I see where you're going.

Right now all of the modules are pretty flat in terms of file structure and the module structure kinda resembles that. With that said, every `directive/provider/service` should belong to a single module. ( @rynomars, I think this addresses your point).

If we're going with this flat structure, no module should have multiple `directives/providers/services` appended to it. Directives are the exception because it has a `controller` (and maybe we can abstract the link method). Can't see the need to add `services` on to a `directive` as they are meant to be shared across multiple components, that breaks encapsulation. The controller (which is an ES6 class) should be enough. It's also similar to how we are structuring out pages in core apps. At this point, we will only have 1 addition file appended to the module (the `directive` `controller`) opposed to an open ended amount.

If we take a look at `accordion` it has 3 other directives (`accordion-header`, `accordion-panel`, `accordion-body`) that require `accordion`. If we were to nest those directives inside of the `accordion` folder and maybe append those onto the `accordion` module opposed to being completely separate, the `.module` would make sense as the module have an opportunity to grow with additional `directives` opposed to every `directive` belonging to a single module.

So the question is, which path to take to keep the module structure resembling the file structure, flat or adding more depth to justify `.module` files.

Obviously, I prefer flat to keep things simple.

**Comment by jsmylnycky** [comment]
@DarrylD @rynomars With all of this said, I'd like to propose this structure:

Keep the folder structure flat like it currently is, how Darryl demonstrated in his most recent post.  Every controller, service, etc will be exactly one module in and of itself.  The exception would be a directive which could also contain a controller, but the files will continue to be separated as is currently.

For the case where you may want to bundle multiple items together (lets use the case of our $peach  providers), another module would then be created that brings them all together (such as how the peach module brings together $peachUser, $peachAccount, etc).



Is everyone in agreement on this proposed structure?  We'll obviously want to be all on board with this, as it will dictate how we construct our projects moving forward.


Also @DarrylD  I mentioned renaming the files so instead of peach.user.js, you just have peach-user.js.... Thinking about it, it's a bit redundant to have the peach prefix, since the other thing already is built for us.  Instead, can you switch it to /providers/user/user.js, etc?  I think the "all encompassing" one should still be 'peach' tho.

**Comment by rynomars** [comment]
@DarrylD @jsmylnycky I am good with that solution.

**Comment by jsmylnycky** [comment]
Great, glad we're all onboard with this solution :)

So Darryl, before we merge, please:

1) Get rid of the peach- and peach. prefixes on the providers.  You can leave the all-encompassing one named 'peach' that brings all of the providers together.

2) Ping me on hipchat and we'll talk real quick on if the directive files themselves should have the .directive added to them.  I'm leaning towards no unless we go through and change this for EVERYTHING, since that would turn all of the peach files into account.provider.js, etc.  I can be persuaded one way or another, but we'll get @rynomars when we're all on at the same time and figure out which way we feel is correct and want to enforce.

**Comment by DarrylD** [comment]
@jsmylnycky 

1) :+1: 

2) I'm ok with how it currently is. I was thinking if we added `.module` adding `.directive` would keep things consistent. Since we're not, that can be disregarded.

**Comment by againsley** [comment]
:+1:  Thanks guys.

@jsmylnycky if you haven't already will you make sure this is doc'ed in our style guide?

**Comment by jsmylnycky** [comment]
@DarrylD  Looks good to me now :+1:  Please go ahead and merge --squash.

@againsley  Yeah, I'll leave this PR open until I get it documented.

**Comment by DarrylD** [comment]
@jsmylnycky 

Tried to get PhantomJS working by adding these modules:
https://github.com/conversocial/phantomjs-polyfill
https://github.com/karma-runner/karma-ng-html2js-preprocessor

Still unable to get those trouble cases to work. 

What I did notice, in karma's "debug mode", if you view the console, you see the same issues. The only difference is that the tests in the terminal says everything is fine while the console in debug mode tab shows the same errors we're getting in PhamtonJS.

screen shot 2015-09-01 at 9 03 05 pm

For that reason, I haven't merged yet, but the pr is updated.

**Comment by jsmylnycky** [comment]
@DarrylD  Go ahead and remove the phantomjs/html2js stuff that was added in.  If the unit tests are working ok with Chrome how they were, that's fine for now.  It's more important for us to get this release out the door to avoid merge becoming a nightmare.

We can work on the phantomjs stuff when we have a little bit more free time.

File: src/services/toast/toast.js [hunk 2]
```diff
@@ -21,4 +23,7 @@ class PeachToastService {
 
 PeachToastService.$inject = ['$mdToast'];
 
-export default PeachToastService;
+angular.module(moduleName, [])
+  .service('peachToastService', PeachToastService);
```
Comment (jsmylnycky) [inline src/services/toast/toast.js]: This service name should be $peachToast

**Comment by DarrylD** [comment]
@jsmylnycky squashed/merged ed887dd3479ac5342b434b823e59c6e79e649684

**Comment by jsmylnycky** [comment]
@DarrylD Thanks! Released v0.4.0

**Pull request closed without merging**